src/fs.rs

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2018 Adgear
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+use std::env::temp_dir;
+use std::fs::{File, remove_file};
+use std::io;
+use std::path::PathBuf;
+
+use rand::Rng;
+
+pub struct TempFile {
+    path: PathBuf,
+}
+
+impl TempFile {
+    pub fn new() -> io::Result<Self> {
+        let mut rng = Rng::new();
+        let mut path = None;
+        for _ in 0..50 {
+            let tempfile = temp_dir()
+                .join(format!("file{}", rng.next()));
+            if !tempfile.exists() {
```

Comment:
It'd be cool if we could pass the prefix so that a temp file for Prophet is called `prophet.1234` and one for ACR is called `acr.6789`.

Reply:
Done.